src/UrlSession.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace iMi\LaravelTransSid;
+
+use Closure;
+
+class UrlSession
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        ini_set('session.use_trans_sid', 1);
+        ini_set('session.use_cookies', 0);
+        ini_set('session.use_only_cookies', 0);
```

Comment:
@rgroli Unfortunately, this breaks tests with the following error message:

```
[2021-03-25 15:24:42] testing.ERROR: ini_set(): Headers already sent. You cannot change the session module's ini settings at this time {"userId":1,"exception":"[object] (ErrorException(code: 0): ini_set(): Headers already sent. You cannot change the session module's ini settings at this time at ./vendor/imi/laravel-transsid/src/UrlSession.php:18)
[stacktrace]
#0 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1  ./vendor/imi/laravel-transsid/src/UrlSession.php(18): ini_set()
#2  ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): iMi\\LaravelTransSid\\UrlSession->handle()
#3  ./vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#4  ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle()
#5  ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(77): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#6  ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle()
#7  ./vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#8  ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle()
#9  ./vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#10 ./vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(63): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest()
#11 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle()
#12 ./vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#13 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle()
#14 ./vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#15 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle()
#16 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#17 ./vendor/laravel/framework/src/Illuminate/Routing/Router.php(693): Illuminate\\Pipeline\\Pipeline->then()
#18 ./vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\\Routing\\Router->runRouteWithinStack()
#19 ./vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\\Routing\\Router->runRoute()
#20 ./vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\\Routing\\Router->dispatchToRoute()
#21 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch()
#22 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}()
#23 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#24 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#25 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#26 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle()
#27 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#28 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle()
#29 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#30 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle()
#31 ./vendor/fruitcake/laravel-cors/src/HandleCors.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#32 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle()
#33 ./vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#34 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle()
#35 ./vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}()
#36 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then()
#37 ./vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter()
#38 ./vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(508): Illuminate\\Foundation\\Http\\Kernel->handle()
#39 ./vendor/laravel/framework/src/Illuminate/Foundation/Testing/Concerns/MakesHttpRequests.php(292): Illuminate\\Foundation\\Testing\\TestCase->call()
#40 ./tests/Feature/BundleSelectTest.php(41): Illuminate\\Foundation\\Testing\\TestCase->get()
#41 ./vendor/phpunit/phpunit/src/Framework/TestCase.php(1536): Tests\\Feature\\BundleSelectTest->testShouldSeeOnlyBundlesWithBudget()
#42 ./vendor/phpunit/phpunit/src/Framework/TestCase.php(1142): PHPUnit\\Framework\\TestCase->runTest()
#43 ./vendor/phpunit/phpunit/src/Framework/TestResult.php(730): PHPUnit\\Framework\\TestCase->runBare()
#44 ./vendor/phpunit/phpunit/src/Framework/TestCase.php(883): PHPUnit\\Framework\\TestResult->run()
#45 ./vendor/phpunit/phpunit/src/Framework/TestSuite.php(669): PHPUnit\\Framework\\TestCase->run()
#46 ./vendor/phpunit/phpunit/src/Framework/TestSuite.php(669): PHPUnit\\Framework\\TestSuite->run()
#47 ./vendor/phpunit/phpunit/src/TextUI/TestRunner.php(667): PHPUnit\\Framework\\TestSuite->run()
#48 ./vendor/phpunit/phpunit/src/TextUI/Command.php(148): PHPUnit\\TextUI\\TestRunner->run()
#49 ./vendor/phpunit/phpunit/src/TextUI/Command.php(101): PHPUnit\\TextUI\\Command->run()
#50 ./vendor/phpunit/phpunit/phpunit(61): PHPUnit\\TextUI\\Command::main()
#51 {main}
"} 
```

Do you have an idea why that might be the case or how to fix it? Why do we even need this?

Reply:
@kolaente I wasn't even aware there were any tests in this repo :) 

You are right, you can remove the lines 18-20 from URLSession. So that's an "empty" middleware which is only used to opt-in to the package funtionality on route/routegroup basis.

Reply:
Oh, there aren't any tests in this repo. This happend in a project which has tests that uses this package.

I've started to refactor this in https://github.com/iMi-digital/laravel-transsid/pull/7, feel free to leave a review :upside_down_face: